pghoard/transfer.py

```diff
@@ -513,6 +539,10 @@ def handle_upload(self, site, key, file_to_transfer: UploadEvent):
 
             # Sleep for a bit to avoid busy looping. Increase sleep time if the op fails multiple times
             self.sleep(min(0.5 * 2 ** (file_to_transfer.retry_number - 1), 20))
-
+            if file_to_transfer.file_type in (
+                FileType.Basebackup, FileType.Basebackup_chunk, FileType.Basebackup_delta, FileType.Basebackup_delta_chunk
+            ) and file_to_transfer.retry_number > 2:
+                persisted_progress = PersistedProgress.read(metrics=self.metrics)
+                persisted_progress.reset(key, metrics=self.metrics)
```

Comment:
We should only reset the statistics when we complete the backup successfully. The OG ticket described what to do in this situation as:

> if current progress <= persisted progress, emit a metric (e.g. basebackup-stalled) whose value is the number of seconds since the persisted epoch.

So basically, we should only ever reset stats on success, otherwise emit a metric to indicate we are stalled and no longer making progress.

Reply:
this should be file_to_transfer.retry_number < 2. If it fails on the first try, it should try again once, and if it fails again, it should never reset.  If one of them fails at 50%, this is tracked in a persisted file, pghoard will retry the file from the beginning, and the hash key remains the same. When checking the progress, the previous state file is 50MB (50%) from the previous write. When comparing it with the current uploaded bytes (maybe 10Mb), it is marked as stalled, just to avoid that.